modules/standard/CommDiagnostics.chpl

```diff
@@ -343,17 +314,7 @@ module CommDiagnostics
   proc getCommDiagnostics() {
     var D: [LocaleSpace] commDiagnostics;
     for loc in Locales do on loc {
-      // See note above regarding extern records
-      D(loc.id).get = chpl_numCommGets();
-      D(loc.id).get_nb = chpl_numCommNBGets();
-      D(loc.id).put = chpl_numCommPuts();
-      D(loc.id).put_nb = chpl_numCommNBPuts();
-      D(loc.id).test_nb = chpl_numCommTestNB();
-      D(loc.id).wait_nb = chpl_numCommWaitNB();
-      D(loc.id).try_nb = chpl_numCommTryNB();
-      D(loc.id).fork = chpl_numCommForks();
-      D(loc.id).fork_fast = chpl_numCommFastForks();
-      D(loc.id).fork_nb = chpl_numCommNBForks();
+      D(loc.id) = getCommDiagnosticsHere();
```

Comment:
Oh if this works that would be awesome -- we've been talking about wanting to do this for ages. Thanks!